lib/spi/types.ts

```diff
@@ -221,4 +223,21 @@ export class ClusterInfo {
         assert(result, `Required resource ${name} is missing.`);
         return result!;
     }
+
+    /**
+     * Update executionContext map
+     * @param addOn
+     * @param Values
+     */
+    public addExecutionContext(addOn: string, values: Values) {
```

Comment:
Since it is specific to add-on, not just any execution context, I suggest renaming to `addAddOnContext` (execution makes sense, just for brevity I thought addon context will do).

Reply:
Agreed - `AddOnContext` is more precise and easier to understand.